rust/private/rust.bzl

```diff
@@ -106,18 +106,21 @@ def _determine_lib_name(name, crate_type, toolchain, lib_hash = None):
         extension = extension,
     )
 
-def get_edition(attr, toolchain):
+def get_edition(attr, toolchain, label):
     """Returns the Rust edition from either the current rule's attirbutes or the current `rust_toolchain`
 
     Args:
         attr (struct): The current rule's attributes
         toolchain (rust_toolchain): The `rust_toolchain` for the current target
+        label (Label): The label of the target being built
 
     Returns:
         str: The target Rust edition
     """
     if getattr(attr, "edition"):
         return attr.edition
+    elif toolchain.default_edition == "":
```

Comment:
I attempted to use `toolchain.default_edition == None`, but `None` is not a valid default value for an attribute of type `attr.string` and I didn't see how to make an optional nullable string attribute. Please let me know whether this seems fine.

Reply:
Could this work?
```suggestion
    elif not toolchain.default_edition:
```